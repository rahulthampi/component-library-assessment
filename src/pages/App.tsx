function App() {
  return (
    <div className="p-4 bg-primary rounded-lg text-center">
      <h1 className="text-display-2xl font-inter text-primary-foreground">Hello world!</h1>
    </div>
  );
}
App.displayName = 'App';

export default App;
