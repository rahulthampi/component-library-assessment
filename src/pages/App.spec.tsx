import { render, screen } from '@testing-library/react';

import App from './App';

describe('App', () => {
  it('renders the App component with ModalWrapper and CardWrapper', () => {
    const wrapper = render(<App />);

    expect(wrapper).toMatchSnapshot();

    // Check if both ModalWrapper and CardWrapper components are rendered.
    const modalWrapper = screen.getByTestId('modal-wrapper');
    const cardWrapper = screen.getByTestId('card-wrapper');

    expect(modalWrapper).toBeInTheDocument();
    expect(cardWrapper).toBeInTheDocument();
  });
});
