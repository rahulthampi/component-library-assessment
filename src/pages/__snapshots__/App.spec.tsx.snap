// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component with ModalWrapper and CardWrapper 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex gap-4 p-4 flex-wrap"
        data-testid="modal-wrapper"
      >
        <button
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
          data-state="closed"
          type="button"
        >
          Open Modal
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[180px]"
        >
          Auto open after 1s
        </button>
      </div>
      <div
        class="flex gap-4 p-4 flex-wrap"
        data-testid="card-wrapper"
      >
        <div
          class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4 w-[385px]"
        >
          <div
            class="flex gap-4 items-center"
          >
            <span
              class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
            >
              <span
                class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
              >
                RT
              </span>
            </span>
            <div
              class="w-full flex flex-col justify-between"
            >
              <span
                class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
              >
                Wallet
              </span>
              <h3
                class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
              >
                Safe
              </h3>
            </div>
          </div>
        </div>
        <div
          class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-4 w-[385px]"
        >
          <div
            class="flex gap-4 items-center"
          >
            <span
              class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
            >
              <span
                class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
              >
                RT
              </span>
            </span>
            <div
              class="w-full flex flex-col justify-between"
            >
              <span
                class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
              >
                Wallet
              </span>
              <h3
                class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
              >
                Safe
              </h3>
            </div>
          </div>
        </div>
        <div
          class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-none hover:bg-neutral p-4 w-[385px]"
        >
          <div
            class="flex gap-4 items-center"
          >
            <span
              class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
            >
              <span
                class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
              >
                RT
              </span>
            </span>
            <div
              class="w-full flex flex-col justify-between"
            >
              <span
                class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
              >
                Wallet
              </span>
              <h3
                class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
              >
                Safe
              </h3>
            </div>
          </div>
        </div>
        <a
          class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-4 w-[385px] flex gap-4 items-center"
          href="#"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
            >
              RT
            </span>
          </span>
          <div
            class="w-full flex flex-col justify-between"
          >
            <span
              class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
            >
              Wallet
            </span>
            <h3
              class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
            >
              Safe
            </h3>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex gap-4 p-4 flex-wrap"
      data-testid="modal-wrapper"
    >
      <button
        aria-controls="radix-:r0:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
        data-state="closed"
        type="button"
      >
        Open Modal
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-[180px]"
      >
        Auto open after 1s
      </button>
    </div>
    <div
      class="flex gap-4 p-4 flex-wrap"
      data-testid="card-wrapper"
    >
      <div
        class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-border hover:bg-neutral p-4 w-[385px]"
      >
        <div
          class="flex gap-4 items-center"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
            >
              RT
            </span>
          </span>
          <div
            class="w-full flex flex-col justify-between"
          >
            <span
              class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
            >
              Wallet
            </span>
            <h3
              class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
            >
              Safe
            </h3>
          </div>
        </div>
      </div>
      <div
        class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-4 w-[385px]"
      >
        <div
          class="flex gap-4 items-center"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
            >
              RT
            </span>
          </span>
          <div
            class="w-full flex flex-col justify-between"
          >
            <span
              class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
            >
              Wallet
            </span>
            <h3
              class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
            >
              Safe
            </h3>
          </div>
        </div>
      </div>
      <div
        class="flex rounded border bg-background transition-colors outline-none focus:border-border-ring border-none hover:bg-neutral p-4 w-[385px]"
      >
        <div
          class="flex gap-4 items-center"
        >
          <span
            class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
            >
              RT
            </span>
          </span>
          <div
            class="w-full flex flex-col justify-between"
          >
            <span
              class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
            >
              Wallet
            </span>
            <h3
              class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
            >
              Safe
            </h3>
          </div>
        </div>
      </div>
      <a
        class="flex rounded border bg-background outline-none focus:border-border-ring border-transparent shadow transition-shadow hover:shadow-md p-4 w-[385px] flex gap-4 items-center"
        href="#"
      >
        <span
          class="relative flex shrink-0 overflow-hidden rounded-full w-12 h-12"
        >
          <span
            class="flex h-full w-full items-center justify-center rounded-full bg-neutral font-semibold text-t-xl"
          >
            RT
          </span>
        </span>
        <div
          class="w-full flex flex-col justify-between"
        >
          <span
            class="rt-Badge uppercase text-t-xs font-bold tracking-wider text-neutral-foreground rt-r-size-1 rt-variant-soft"
          >
            Wallet
          </span>
          <h3
            class="rt-Heading font-inter font-semibold text-t-lg rt-r-size-6 rt-r-weight-bold"
          >
            Safe
          </h3>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
